tests/test_utilities.py

```diff
@@ -35,10 +52,34 @@
 from parlant.core.guideline_tool_associations import GuidelineToolAssociationStore
 from parlant.core.guidelines import Guideline, GuidelineStore
 from parlant.core.logging import LogLevel, Logger
-from parlant.core.sessions import Event, MessageEventData, Session, SessionId, SessionStore
+from parlant.core.nlp.generation import (
+    FallbackSchematicGenerator,
+    GenerationInfo,
+    SchematicGenerationResult,
+    SchematicGenerator,
+    UsageInfo,
+)
+from parlant.core.nlp.tokenization import EstimatingTokenizer
+from parlant.core.sessions import (
+    _GenerationInfoDocument,
+    _UsageInfoDocument,
+    Event,
+    MessageEventData,
+    Session,
+    SessionId,
+    SessionStore,
+)
 from parlant.core.tools import LocalToolService, ToolId, ToolResult
+from parlant.core.persistence.common import ObjectId
+from parlant.core.persistence.document_database import DocumentCollection
 
 T = TypeVar("T")
+GLOBAL_CACHE_FILE = Path("_schematic_cache.json")
```

Comment:
Not standard form to name the file with an underscore prefix. Files aren't python attributes :)
Let's call it `schematic_generation_cache.json`